integration-tests/integration_test.go

```diff
@@ -0,0 +1,986 @@
+							},
+						)
+						require.NoError(t, err, "Should write duplicate request")
+
+						ctx, cancel := context.WithTimeout(
+							context.Background(),
+							responseWaitDuration,
+						)
+						defer cancel()
+
+						message, err := reader.ReadMessage(ctx)
+						require.NoError(t, err, "Should read duplicate response")
+
+						resultSet, err := avroSchema.DeserializeRedeemResultSet(
+							bytes.NewReader(message.Value),
+						)
+						require.NoError(t, err, "Should deserialize duplicate response")
+
+						require.Equal(t, duplicateRequestID, resultSet.Request_id,
+							"Duplicate response ID should match")
+
+						require.NotEmpty(t, resultSet.Data, "Should have duplicate results")
+
+						result := resultSet.Data[0]
+						assert.Equal(t,
+							avroSchema.RedeemResultStatusDuplicate_redemption,
+							result.Status,
+							"Duplicate redemption should be detected")
+					})
+				})
+
+				// Only need to run the detailed tests once
+				break
+			}
+		}
+	})
+}
+
+func TestIssuerV1(t *testing.T) {
+	client := &http.Client{Timeout: 30 * time.Second}
+
+	exp := time.Now().Add(time.Hour).UTC()
+
+	issuerRequest := issuerV3CreateRequest{
+		Name:      "TestIssuer-" + uuid.New().String(),
+		Cohort:    1,
+		MaxTokens: 1,
+		ExpiresAt: &exp,
+	}
+
+	t.Run("create_issuer", func(t *testing.T) {
+		body, err := json.Marshal(issuerRequest)
+		require.NoError(t, err)
+
+		req, err := http.NewRequest(
+			http.MethodPost,
+			"http://cbp:2416/v1/issuer",
+			bytes.NewBuffer(body),
+		)
+		require.NoError(t, err)
+
+		resp, err := client.Do(req)
+		require.NoError(t, err)
+
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+	})
+
+	t.Run("get_issuer", func(t *testing.T) {
+		t.Log("TESTING HTTP ISSUER GET ENDPOINT")
+
+		t.Logf("Retrieving issuer information for '%s'...", issuerRequest.Name)
+		req, err := http.NewRequest(
+			"GET",
+			fmt.Sprintf("http://cbp:2416/v1/issuer/%s", issuerRequest.Name),
+			nil,
+		)
+		require.NoError(t, err, "failed to create GET issuer HTTP request")
+
+		client := &http.Client{Timeout: 30 * time.Second}
+		resp, err := client.Do(req)
+		require.NoError(t, err, "failed to make GET issuer HTTP request")
+		defer resp.Body.Close()
+
+		t.Logf("Issuer GET response status: %s", resp.Status)
+		require.Equal(
+			t,
+			http.StatusOK,
+			resp.StatusCode,
+			"GET issuer request should succeed",
+		)
+
+		var issuerResp issuerResponse
+		err = json.NewDecoder(resp.Body).Decode(&issuerResp)
+		require.NoError(t, err, "failed to decode issuer response")
+
+		t.Logf("Issuer details: ID=%s, Name=%s, Cohort=%d",
+			issuerResp.ID, issuerResp.Name, issuerResp.Cohort)
+
+		assert.NotEmpty(t, issuerResp.ID, "issuer ID should not be empty")
+		assert.Equal(t, issuerRequest.Name, issuerResp.Name, "issuer name should match")
+		assert.NotNil(t, issuerResp.PublicKey, "public key should not be nil")
+		assert.Equal(t, int32(1), issuerResp.Cohort, "cohort should be 1")
+
+		if issuerResp.ExpiresAt != "" {
```

Comment:
I don't think we need this block of code. The only condition that should ever be true is that the issuer has an expires at equal to what we set at the beginning of the test [here](https://github.com/brave-intl/challenge-bypass-server/pull/816/files#diff-b33316820a7d77030ab0e3e5b0ef570e245b1d84a5326451277241084b6e7f45R376), any other value should fail.

We can simply assert that the value is equal using `assert.Equal(t, issReq.ExpiresAt.Format(time.RFC3339), actual.ExpiresAt)` which is a much stronger assertion than simply in the future and anything else should fail.